### PR TITLE
feat($sanitize): Added support for the sftp protocol in $$ SanitizeUriProvider and linky filter

### PR DESCRIPTION
**Pull request by XFree**
… SanitizeUriProvider

Added support for the sftp protocol in $$ SanitizeUriProvider and linky filter

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**
sftp not suported

**What is the new behavior (if this is a feature change)?**
sftp suported

**Does this PR introduce a breaking change?**
no

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

**Comment by XFree** [comment]
I do not understand why the build is falling

**Comment by Narretz** [comment]
Hi, the build error is unrelated to your PR. We are looking into it.

**Comment by XFree** [comment]
@Narretz, Tell me, where are the rules for adding to the documentation about the fixes.

**Comment by Narretz** [comment]
@XFree the docs are in the code files themselves. However, it looks like the default whitelist is not documented. The docs for linky are here: https://github.com/angular/angular.js/blob/master/src/ngSanitize/filter/linky.js

Btw, the scope for this should be `feat($sanitize) ...` as `$$SanitizeUriProvider` is a private service.

**Comment by Narretz** [comment]
Could you please add `(s)ftp` tests to linky and info the linky docs that sftp is supported?

**Comment by XFree** [comment]
Is the documentation generated automatically?
I did not really understand how to properly arrange the documentation.

**Comment by Narretz** [comment]
@XFree the docs are created from the comments around the functions, e.g. https://github.com/angular/angular.js/blob/ecc09a45f02454e4e2478ad22cd0d179dccc7916/src/ngSanitize/filter/linky.js#L8-L10

**Comment by XFree** [comment]
@Narretz , Oh. Thank you.

**Review by gkalpak — APPROVED** [review]

**Comment by Narretz** [comment]
@mprobst / @rjamet could one of you have a quick look at this security related PR? It's very straightforward, I just wanna follow protocol. :)

**Comment by rjamet** [comment]
Is there any browser with built-in support of sftp?... 

That looks safe, and so I'm okay with merging, but I think that might be better suited as an additional config knob that gets set in the application itself. 

SFTP matches the idea of fetching stuff at the end of a link, so I think it's reasonable. But if we allow sftp, then you could push that logic to ssh://, magnet://, onion:// and a more which would be surprising for devs who didn't explicitly allow those. In all of those cases, I'd rather see something like linkyProvider.allowProtocols("sftp://") / $compileProvider.allowProtocols("sftp://") in the config phase of the application: this way, you don't change the default, and if you need to use these protocols, then two lines are enough to enable them app-wide.

**Comment by XFree** [comment]
@rjamet , In this case, when this need to do this for https, but this is beyond the scope of this PR.

File: src/ng/sanitizeUri.js [hunk 1]
```diff
@@ -6,8 +6,8 @@
  * Private service to sanitize uris for links and images. Used by $compile and $sanitize.
  */
 function $$SanitizeUriProvider() {
-  var aHrefSanitizationWhitelist = /^\s*(https?|ftp|mailto|tel|file):/,
-    imgSrcSanitizationWhitelist = /^\s*((https?|ftp|file|blob):|data:image\/)/;
+  var aHrefSanitizationWhitelist = /^\s*(https?|s?ftp|mailto|tel|file):/,
+    imgSrcSanitizationWhitelist = /^\s*((https?|s?ftp|file|blob):|data:image\/)/;
```
Comment (rjamet) [inline src/ng/sanitizeUri.js]: Oops, I missed this, sorry: do you need to update imgSrc too ? I can see how navigational links could work, but if browsers don't support the protocol, images and such wouldn't work at all (Chrome seems to complain with `Failed to load resource: net::ERR_UNKNOWN_URL_SCHEME` at least).

Comment (XFree) [reply src/ng/sanitizeUri.js]: I'm fix it.

**Comment by rjamet** [comment]
LGTM security-wise, I'll let the Angular team decide for the rest.

**Comment by XFree** [comment]
I do not understand what the problems are with CI?

**Comment by gkalpak** [comment]
`commitplease` was unable to validate the commit messages (didn't really looked into why). Given that this is the only test that failed, I think it is fine to ignore the CI failure.

**Comment by XFree** [comment]
When will Pull request be merged? =)

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**